### PR TITLE
[nrf noup] ci: Updated test-spec.yml for Cloud CI testing

### PR DESCRIPTION
**Pull request by tony-le-24**
Added an entry for CI-cloud-test that will run the downstream job of test-fw-nrfconnect-nrf-iot_cloud when any of the listed modules change.

See IRIS-5423

Signed-off-by: Tony Le <tony.le@nordicsemi.no>

### CODE REVIEW

**Comment by glarsennordic** [comment]
I think a special cherry picking and naming process needs to be followed for commits to this repository

See the previous commits to this file:

https://github.com/nrfconnect/sdk-zephyr/commits/main/.github/test-spec.yml

**Review by glarsennordic — CHANGES_REQUESTED** [review]
see prior comment

**Review by glarsennordic — APPROVED** [review]
LGTM

File: .github/test-spec.yml [hunk 1]
```diff
@@ -125,3 +125,19 @@
...
   - "subsys/net/**/*"
   - "subsys/settings/**/*"
+
+"CI-cloud-test":
+  - "include/zephyr/dfu/**/*"
+  - "include/zephyr/net/**/*"
+  - "include/zephyr/posix/**/*"
+  - "include/zephyr/settings/**/*"
+  - "drivers/led/**/*"
+  - "drivers/net/**/*"
```
Comment (plskeggs) [inline .github/test-spec.yml:135]: drivers/net seems logical to include based on the name, but we don't use any of those specific drivers

**Review by plskeggs — APPROVED** [review]
I suspect these are overly-broad rules, and will trigger more runs of our tests than really necessary.  But since this mostly will happen during upmerges and not day-to-day, hopefully it will be fine.

**Review by shantha-14 — APPROVED** [review]

**Pull request merged**